src/core/lib/iomgr/work_serializer.cc

```diff
@@ -55,36 +73,53 @@ void WorkSerializer::WorkSerializerImpl::Run(
     gpr_log(GPR_INFO, "WorkSerializer::Run() %p Scheduling callback [%s:%d]",
             this, location.file(), location.line());
   }
-  const size_t prev_size = size_.fetch_add(1);
+  const uint64_t prev_ref_pair =
+      refs_.fetch_add(MakeRefPair(1, 1), std::memory_order_acq_rel);
   // The work serializer should not have been orphaned.
-  GPR_DEBUG_ASSERT(prev_size > 0);
-  if (prev_size == 1) {
-    // There is no other closure executing right now on this work serializer.
-    // Execute this closure immediately.
+  GPR_DEBUG_ASSERT(GetSize(prev_ref_pair) > 0);
+  // If there is no other callback executing right now on this work serializer,
```

Comment:
This comment is in an odd place and seems a little misleading, since there is no lock here.

I suggest removing this comment and instead adding a comment on line 76 above that says:

"""
Increment queue size for the new callback and owner count to attempt to take ownership of the WorkSerializer.
"""

Reply:
Done.